pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -375,7 +376,7 @@ func (c *DeploymentConfigController) handleErr(err error, key interface{}) {
 		return
 	}
 
-	if c.queue.NumRequeues(key) < 10 {
+	if c.queue.NumRequeues(key) < MaxRetries {
 		c.queue.AddRateLimited(key)
 	} else {
 		glog.V(2).Infof(err.Error())
```

Comment:
I prefer your upstream structure with the reporting on retry, return, then util.HandleErr, forget.